### PR TITLE
`test_context`: Replace `env_var`/`env_vars` with `monkeypatch.setenv`

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Replace `env_var` usage in `tests/base/test_context.py with `monkeypatch.setenv`.

Xref https://github.com/conda/conda/issues/14095

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](https://github.com/conda/conda/blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: tests/base/test_context.py [hunk 1]
```diff
@@ -265,16 +259,15 @@ def test_custom_multichannels(testdata: None):
...
-        )
+def test_restore_free_channel(monkeypatch: MonkeyPatch) -> None:
+    free_channel = "https://repo.anaconda.com/pkgs/free"
+    assert free_channel not in context.default_channels
+
+    monkeypatch.setenv("CONDA_RESTORE_FREE_CHANNEL", "true")
+    reset_context()
+    assert context.restore_free_channel
+
+    assert context.default_channels[1] == free_channel
```
Comment (ForgottenProgramme) [inline tests/base/test_context.py:270]: 👍🏼

File: tests/base/test_context.py [hunk 2]
```diff
@@ -330,114 +321,112 @@ def test_context_parameters_have_descriptions(testdata: None):
...
+    (envs2 := path_factory()).mkdir()
+    envs_dirs = (str(envs1), str(envs2))
+
+    monkeypatch.setenv("CONDA_ENVS_DIRS", os.pathsep.join(envs_dirs))
+    reset_context()
+    assert context._envs_dirs == envs_dirs
+    assert context.envs_dirs[:2] == envs_dirs
+
+    # with both dirs writable, choose first
+    reset_context(argparse_args=SimpleNamespace(name="blarg"))
```
Comment (ForgottenProgramme) [inline tests/base/test_context.py:359]: Can `SimpleNamespace` replace `AttrDict` in all places in this file?

**Comment by dholth** [comment]
Why do we need to increase lines-of-code and why shoul we care about the indentation?

**Comment by jezdez** [comment]
> Why do we need to increase lines-of-code? Why should we care about the indentation?

I think this is to eventually remove the custom bits in the test suite?

**Review by jezdez — APPROVED** [review]

**Comment by kenodegard** [comment]
The last use case of `env_var`/`env_vars` within the conda source code is slated for deprecation in `conda 25.3` (see [`conda.plan.install_actions`](https://github.com/conda/conda/blob/aa0fb6f3ae669a5ade575d340555aa6a9f71ef5e/conda/plan.py#L429-L452)). Otherwise it is used in our test suite and in a small handful of downstream locations (e.g., conda-libmamba-solver and conda-build).

I don't see a strong argument for keeping this helper function around when we can accomplish the same thing with `monkeypatch`.

**Pull request merged**